src/frame.rs

```diff
@@ -1130,6 +1137,20 @@ pub fn encode_stream_header(
     Ok(())
 }
 
+pub fn encode_dgram_header(length: u64, b: &mut octets::OctetsMut) -> Result<()> {
+    let mut ty: u8 = 0x30;
+
+    // Always encode length
+    ty |= 0x01;
+
+    b.put_varint(u64::from(ty))?;
```

Comment:
The hope with the old code was that we might go back one day and find a smart way to decide if we will include the datagram length or not.

The changes in this PR seem to be taking the decision to include length further, which is OK. I wonder if we should just use 0x31 type directly.

Reply:
This is similar to how `STREAM` frames are encoded as well. The original idea was to set the flags separately for clarity, so e.g. we can have a comment to describe what's going on.

Reply:
oh yeah, in that case its fine